src/System.IO.Pipelines/src/System/IO/Pipelines/Pipe.DefaultPipeWriter.cs

```diff
@@ -26,19 +28,19 @@ public DefaultPipeWriter(Pipe pipe)
 
             public override void OnReaderCompleted(Action<Exception, object> callback, object state) => _pipe.OnReaderCompleted(callback, state);
 
-            public override PipeAwaiter<FlushResult> FlushAsync(CancellationToken cancellationToken = default) => _pipe.FlushAsync(cancellationToken);
+            public override ValueTask<FlushResult> FlushAsync(CancellationToken cancellationToken = default) => _pipe.FlushAsync(cancellationToken);
 
             public override void Advance(int bytes) => _pipe.Advance(bytes);
 
             public override Memory<byte> GetMemory(int sizeHint = 0) => _pipe.GetMemory(sizeHint);
 
             public override Span<byte> GetSpan(int sizeHint = 0) => _pipe.GetMemory(sizeHint).Span;
 
-            public bool IsCompleted => _pipe.IsFlushAsyncCompleted;
+            public ValueTaskSourceStatus GetStatus(short token) => _pipe.GetFlushAsyncGetStatus();
 
-            public FlushResult GetResult() => _pipe.GetFlushAsyncResult();
+            public FlushResult GetResult(short token) => _pipe.GetFlushAsyncResult();
```

Comment:
Ignoring the token here is by design, correct?

Reply:
Yep, the implementation can choose.